plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -100,11 +101,9 @@ private static void generateFilterPql(StringBuilder pqlBuilder, PinotTableHandle
         ImmutableList.Builder<String> conjunctsBuilder = ImmutableList.builder();
         timePredicate.ifPresent(conjunctsBuilder::add);
         if (!tupleDomain.equals(TupleDomain.all())) {
-            for (PinotColumnHandle columnHandle : columnHandles) {
-                Domain domain = tupleDomain.getDomains().get().get(columnHandle);
-                if (domain != null) {
-                    conjunctsBuilder.add(toPredicate(columnHandle.getColumnName(), domain));
-                }
+            Map<ColumnHandle, Domain> domains = tupleDomain.getDomains().orElseThrow();
```

Comment:
the `f (!tupleDomain.equals(TupleDomain.all())) {` above is redundant (to be followed)